src/compas/utilities/itertools_.py

```diff
@@ -273,14 +278,60 @@ def random_combination_with_replacement(iterable, r):
     return tuple(pool[i] for i in indices)
 
 
+def iterable_like(target, reference, fillvalue=None, as_single=False):
+    """
+    Creates an iterator from a reference object with size equivalent to that of a target iterable.
+
+    Values will be yielded one at a time until the target iterable is exhausted.
+    If target and reference are of uneven size, fillvalue will be used to 
+    substitute the missing values.
+
+    Parameters
+    ----------
+    target : iterable
+        An iterable to be matched in size.
+    reference: object
+        Object taken as departure point.
+    fillvalue : object, optional
+        Fill value. Defaults to `None`.
+    is_single : bool, optional
+        Reference should be regarded as a single entry. Defaults to `True`.
+
+    Returns
+    -------
+    object
+        The next value in the iterator
+
+    Examples
+    --------
+    >>> keys = [0, 1, 2]
+    >>> color = (255, 0, 0)
+    >>> [_ for _ in iterable_like(keys, color)]
+    [255, 0, 0]
+    >>> [_ for _ in iterable_like(keys, color, as_single=True)]
+    [(255, 0, 0), None, None]
+    >>> [_ for _ in iterable_like(keys, color, color, as_single=True)]
+    [(255, 0, 0), (255, 0, 0), (255, 0, 0)]
+    """
+    if as_single:
+        reference = [reference]
+
+    zipped = zip_longest(target, reference, fillvalue=fillvalue)
+    for _ in target:
+        yield next(zipped)[1]
```

Comment:
why not do
```python
for t, r in zip_longest(target, reference, fillvalue=fillvalue):
    yield r
```

Reply:
Because, if for some reason `target` is shorter than `reference`, would instead keep returning values until `reference` is exhausted. 

For example:

```
target = range(2)
reference = ['a', 'b', 'c']

for t, r in zip_longest(target, reference, fillvalue=fillvalue):
    yield r

# yields 'a', 'b', 'c'
```

Thus, I find the current proposed way useful not only to extend but also to _cap_ to a target iterable.